op-node/eth/types.go

```diff
@@ -236,6 +238,33 @@ type PayloadAttributes struct {
 	GasLimit *Uint64Quantity `json:"gasLimit,omitempty"`
 }
 
+type L2BatchJustification struct {
+	PrevBatchLastBlock espresso.Header `json:"prevBatchLastBlock"`
+	FirstBlock espresso.Header `json:"firstBlock"`
+
+	// The block number of `FirstBlock`. This allows us to authenticate the `FirstBlock` header by
+	// computing the commitment of `FirstBlock` and checking it against the commitment at position
+	// `FirstBlockNumber` in a certified sequence (e.g. the sequencer contract). We can also
+	// authenticate `PrevBatchLastBlock` in a similar way, since it is required to be the block
+	// immediately before `FirstBlock`.
+	FirstBlockNumber uint64 `json:"firstBlockNumber"`
+
+	Payload *L2BatchPayloadJustification `json:"payload,omitempty",rlp:"nil"`
```

Comment:
Why have this sub-struct? Might be cleaner to keep things flat if possible

Reply:
I should add a comment. It's because this sub-struct is optional, it's omitted in the various exception cases where we skip HotShot blocks that don't satisfy the constraints

Reply:
ok sounds good